src/services/navigationBar.ts

```diff
@@ -176,23 +190,36 @@ namespace ts.NavigationBar {
             }
         }
 
+        function hasNonAnonymousFunctionDeclarations(nodes: NodeArray<any>) {
+            if (forEach(nodes, s => s.kind === SyntaxKind.FunctionDeclaration && !isEmpty((<FunctionDeclaration>s).name.text))) {
+                return true;
+            }
+        }
+
         function isTopLevelFunctionDeclaration(functionDeclaration: FunctionLikeDeclaration) {
             if (functionDeclaration.kind === SyntaxKind.FunctionDeclaration) {
                 // A function declaration is 'top level' if it contains any function declarations 
                 // within it. 
                 if (functionDeclaration.body && functionDeclaration.body.kind === SyntaxKind.Block) {
                     // Proper function declarations can only have identifier names
-                    if (forEach((<Block>functionDeclaration.body).statements,
-                        s => s.kind === SyntaxKind.FunctionDeclaration && !isEmpty((<FunctionDeclaration>s).name.text))) {
+                    if (hasNonAnonymousFunctionDeclarations((<Block>functionDeclaration.body).statements)) {
 
                         return true;
                     }
 
-                    // Or if it is not parented by another function.  i.e all functions
-                    // at module scope are 'top level'.
+                    // Or if it is not parented by another function(except for parent functions that
```

Comment:
Yes, it is not an ideal solution here, because I check twice here for method declaration and constructor in `isFunctionLikeKind`. Though I think the only other alternative is to rewrite the logic in `isFunctionBlock` and exclude method declarations and constructors.